kore/src/Kore/ModelChecker/Step.hs

```diff
@@ -115,34 +119,34 @@ transitionRule
     -- ^ Map from symbol IDs to defined functions
     -> Prim (CommonModalPattern level) (RewriteRule level Variable)
     -> CommonProofState level
-    -> Transition
-        (CommonProofState level)
+    -> BMCTransition (CommonProofState level)
 transitionRule
     tools
     predicateSimplifier
     patternSimplifier
     axiomSimplifiers
     strategyPrim
     proofState
-  =
-    case strategyPrim of
-        CheckProofState -> transitionCheckProofState proofState
-        Simplify -> transitionSimplify proofState
-        Unroll goalrhs -> transitionUnroll goalrhs proofState
-        ComputeWeakNext rewrites -> transitionComputeWeakNext rewrites proofState
+  = do
+    execState <- Monad.Trans.lift State.get
+    case execState of
```

Comment:
I think this would be clearer, with less creeping indentation:
```.hs
  = do
    execState <- Monad.Trans.lift State.get
    Foldable.traverse_ (const empty) execState  -- End early if any unprovable state was reached
    case strategyPrim of
        ...
```
You could also use a `Bool` instead of `Maybe ()` if you don't want to return the unprovable state itself, then it would look like
```.hs
  = do
    unprovable <- Monad.Trans.lift State.get
    Control.Monad.when unprovable empty  -- End early if any unprovable state was reached.
    case strategyPrim of
        ...
```

Reply:
done